sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsCatalogOptions.java

```diff
@@ -52,4 +54,20 @@
   default String extractCatalog(CaseInsensitiveStringMap options) {
     return CatalogManager.SESSION_CATALOG_NAME();
   }
+
+  /**
+   * Return a Timestamp String given DataFrameReader options.
+   * If this is implemented by data source, then extractTimeTravelVersion can't be implemented.
```

Comment:
```suggestion
   * Extracts the timestamp string for time travel from the given options.
```

Reply:
It's a user error if both timestamp and version are provided in the options. The implementation doesn't need to care about it. Spark should check this and give a clear error message to the users.